Inputmanager now no longer has ownership over the commands

LeeVangraefschepe opened the pull request:
- Removed all command logic from the scene
- Created FreeCamMovement component
- Created WindowManager component

LeeVangraefschepe committed:
Inputmanager now no longer has ownership over the commands

- Removed all command logic from the scene
- Created FreeCamMovement component
- Created WindowManager component

AtlantiaKing reviewed (changes requested):
Command system is nicely done.
Camera pitch doesn't work anymore when looking in the x axis direction.

LeeVangraefschepe committed:
Updated .gitignore file for user settings in visual studio

LeeVangraefschepe committed:
Name & tag of gameobject now return char*

The cached variable was a char* and the function didn't compile when used.

LeeVangraefschepe committed:
Gameobject now deletes itself from parent when SetParent is called

LeeVangraefschepe committed:
Fixed gimbal lock with camera

When freecammovement is added it adds an extra parent to it. This one does the Y rotation. While the camera itself does the X rotation. The parent also handels the position of the camera.
